### DISCUSSION TITLE
Python script tagging album name - find, fetch and add album name and album cover based on Artist/Title

### DISCUSSION THREAD

**Discussion by pawelzet** [AzuraCast/AzuraCast#7256] [Show and tell]
This Python script helps you find, fetch, and add album names and album covers to your music files based on the artist and title. It uses the Spotify API, so you will need to [create a Spotify account and obtain an API key](https://developer.spotify.com). 

Once you have your API key, point the script to the folder containing your music files. The script will then automatically update your music files with the correct album information and cover art.

[Script](https://github.com/pawelzet/album-name-and-cover-fetcher/tree/main)

**Comment by Moonbase59** [comment]
Herectic remark: Why use Spotify when you can tag your files correctly and use the embedded covers or those from MusicBrainz’ Cover Art Archive? _Much_ more precise than guessing from _artist_ and _title_… And free.

Nothing against Python, though. Script is well done.

**Reply by pawelzet** [reply]
@Moonbase59 Spotify API is free to use. Actually I found it more precisely than musicbrainz - tested on both APIs. As I can see for now 99% of the tracks were tagged correctly with their Album names and Embedded Covers from Spotify API. Saved me a lot of time doing it manually. Also Spotify API is much faster. You should try it on first😀

**Reply by unudelaradio** [reply]
It would be nice if these Spotify and MusicBrainz API alternatives will be introduced in AzuraCast Services Settings, under Check Web Services for Album Art for "Now Playing" Tracks. Unfortunately Last.fm is not accurate, most of the times, and these 2 new API alternatives would be very useful for non stop Live Streaming stations. As i mentioned here: https://github.com/AzuraCast/AzuraCast/discussions/7223#discussioncomment-9846838

**Reply by pawelzet** [reply]
@unudelaradio It could be a nice feature to get the album art based on other APIs but what my script does is basically getting album info and album art based on Artist and Title, the LastFM API in Azuracast is fetching Album Cover Art based on Album Title tag, so if you have wrong Album Title it won't work or will fetch wrong cover art.
